<launch>
    <arg name="robot1_active" default="true" />
    <arg name="robot2_active" default="false" />
    <arg name="robot1_tracker" default="trackerC" />
    <arg name="robot2_tracker" default="trackerE" />
    <arg name="side" default="b" />
    <arg name="ekf_" default="false" />
    <arg name="ekf_count_" default="100" />
    <arg name="tole_ekf" default="0.15" />
    <arg name="tole_ekf_angle" default="15" />
    <arg name="frequency" default="100" />  <!--the
    publishing freqency is this value divided by 5-->
    <arg name="print_unit" default="100" />
    <arg name="print_world" default="false" />

    <group ns="robot1" if="$(arg robot1_active)">
        <include file="$(find vive)/launch/vive_world.launch">
            <arg name="survive1_active" value="$(arg robot1_active)" />
            <arg name="survive1_tracker" value="$(arg robot1_tracker)" />
            <arg name="freq_" value="$(arg frequency)" />
            <arg name="unit_" value="$(arg print_unit)" />
            <arg name="side_" value="$(arg side)" />
            <arg name="print_info" value="$(arg print_world)" />
        </include>
        <node name="vive_trackerpose" pkg="vive" type="vive_trackerpose" output="screen">
            <param name="robot_name" type="string" value="robot1" />
            <param name="survive_prefix" type="string" value="survive1_" />
            <param name="tracker" type="string" value="$(arg robot1_tracker)" />
            <param name="ekf_active" type="bool" value="$(arg ekf_)" />
            <param name="ekf_count" type="int" value="$(arg ekf_count_)" />
            <param name="tole_with_ekf" type="double" value="$(arg tole_ekf)" />
            <param name="tole_with_ekf_angle" type="double" value="$(arg tole_ekf_angle)" />
            <param name="freq" type="double" value="$(arg frequency)" />
            <param name="unit" type="double" value="$(arg print_unit)" />
            <param name="rot_from_tracker_W" type="double" value="0" />
            <param name="rot_from_tracker_X" type="double" value="1" />
            <param name="rot_from_tracker_Y" type="double" value="0" />
            <param name="rot_from_tracker_Z" type="double" value="0" />
            <param name="covariance0" type="double" value="0.0001" />
            <param name="covariance7" type="double" value="0.0001" />
            <param name="covariance14" type="double" value="0.0001" />
            <param name="covariance21" type="double" value="0.0001" />
            <param name="covariance28" type="double" value="0.0001" />
            <param name="covariance35" type="double" value="0.0001" />
        </node>
        <!-- <node pkg="vive" type="fake_ekf" name="fake_ekf">
            <param name="fake_ekf_x" type="double" value="1.48"/>
            <param name="fake_ekf_y" type="double" value="0.21"/>
        </node> -->
    </group>

    <group ns="robot2" if="$(arg robot2_active)">
        <include file="$(find vive)/launch/vive_world.launch">
            <arg name="survive2_active" value="$(arg robot2_active)" />
            <arg name="survive2_tracker" value="$(arg robot2_tracker)" />
            <arg name="freq_" value="$(arg frequency)" />
            <arg name="unit_" value="$(arg print_unit)" />
            <arg name="side_" value="$(arg side)" />
            <arg name="print_info" value="$(arg print_world)" />
        </include>
        <node name="vive_trackerpose" pkg="vive" type="vive_trackerpose" output="screen">
            <param name="robot_name" type="string" value="robot2" />
            <param name="survive_prefix" type="string" value="survive2_" />
            <param name="tracker" type="string" value="$(arg robot2_tracker)" />
            <param name="ekf_active" type="bool" value="$(arg ekf_)" />
            <param name="ekf_count" type="int" value="$(arg ekf_count_)" />
            <param name="tole_with_ekf" type="double" value="$(arg tole_ekf)" />
            <param name="tole_with_ekf_angle" type="double" value="$(arg tole_ekf_angle)" />
            <param name="freq" type="double" value="$(arg frequency)" />
            <param name="unit" type="double" value="$(arg print_unit)" />
            <param name="rot_from_tracker_W" type="double" value="0" />
            <param name="rot_from_tracker_X" type="double" value="1" />
            <param name="rot_from_tracker_Y" type="double" value="0" />
            <param name="rot_from_tracker_Z" type="double" value="0" />
            <param name="covariance0" type="double" value="0.001" />
            <param name="covariance7" type="double" value="0.001" />
            <param name="covariance14" type="double" value="0.001" />
            <param name="covariance21" type="double" value="0.001" />
            <param name="covariance28" type="double" value="0.001" />
            <param name="covariance35" type="double" value="0.001" />
        </node>
        <!-- <node pkg="vive" type="fake_ekf" name="fake_ekf">
            <param name="fake_ekf_x" type="double" value="1.48" />
            <param name="fake_ekf_y" type="double" value="0.21" />
        </node> -->
    </group>

</launch>
